### ISSUE TITLE
Upgrade to didkit/ssi v0.4.

### ISSUE THREAD

**Issue by cobward** [spruceid/kepler#90]
Upgrading to didkit and ssi v0.4 leads to a compilation error for libp2p-noise:
```
error[E0282]: type annotations needed
   --> /home/jward/.cargo/registry/src/github.com-1ecc6299db9ec823/libp2p-noise-0.32.0/src/protocol/x25519.rs:221:45
    |
221 |         curve25519_sk.copy_from_slice(&hash.as_ref()[..32]);
    |                                        -----^^^^^^--
    |                                        |    |
    |                                        |    cannot infer type for type parameter `T` declared on the trait `AsRef`
    |                                        this method call resolves to `&T`
    |
    = note: type must be known at this point

For more information about this error, try `rustc --explain E0282`.
error: could not compile `libp2p-noise` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by sbihel** [comment]
FYI it's unrelated to didkit, you can check the tree of dependencies with `cargo tree`

**Comment by chunningham** [comment]
closed by #83 as part of the `spruceid/rust-ipfs` update
